sql/moz-fx-data-shared-prod/search_derived/search_clients_daily_v9/query.sql

```diff
@@ -0,0 +1,333 @@
+      -- MARK: is this the right way to get follow on searches?
+      COUNTIF((is_tagged = TRUE)
+        AND (search_access_point = 'follow_on_from_refine_on_incontent_search'
+        OR search_access_point = 'follow_on_from_refine_on_SERP')) AS serp_follow_on_searches_tagged_count,
+      COUNTIF(is_tagged = FALSE AND num_ads_visible > 0) AS serp_with_ads_organic_count,
+      COUNTIF(is_tagged = TRUE AND num_ads_visible > 0) AS serp_with_ads_tagged_count,
+      SUM(IF(is_tagged = FALSE, num_ad_clicks, 0)) AS serp_ad_clicks_organic_count,
+      SUM(IF(is_tagged = TRUE, num_ad_clicks, 0)) AS serp_ad_clicks_tagged_count,
+      SUM(num_ad_clicks) AS serp_ad_click_sum,
+      ARRAY_AGG(ad_blocker_inferred IGNORE NULLS) AS serp_ad_blocker_inferred,
+      ARRAY_AGG(ad_click_target IGNORE NULLS) as serp_ad_clicks_target,
+      ARRAY_AGG(impression_id IGNORE NULLS) AS serp_impression_id,
+      --this is glean_metrics stuff getting aggregated
+      SUM(browser_engagement_active_ticks / (3600 / 5)) AS active_hours_sum,
+      -- This is commented out because we're not yet sure if we need it (line 190)
+      -- SUM(
+      --   TIMESTAMP_DIFF(
+      --     SAFE_PARSE_TIMESTAMP(subsession_end_time),
+      --     SAFE_PARSE_TIMESTAMP(subsession_start_time),
+      --     SECOND
+      --     ) / NUMERIC '3600'
+      --   ) AS subsession_hours_sum,
+    COUNTIF(subsession_counter = 1) AS sessions_started_on_this_day,
+    SUM(
+      browser_engagement_tab_open_event_count
+    ) AS scalar_parent_browser_engagement_tab_open_event_count_sum,
+    SUM(
+      browser_engagement_uri_count
+    ) AS scalar_parent_browser_engagement_total_uri_count_sum,
+    MAX(
+      browser_engagement_max_concurrent_tab_count
+    ) AS max_concurrent_tab_count_max,
+    MAX(UNIX_DATE(DATE(SAFE_PARSE_TIMESTAMP(string(subsession_start_date))))) - MAX(
+      profile_creation_date
+    ) AS profile_age_in_days,
+    mozfun.map.mode_last(
+      ARRAY_CONCAT_AGG(ping_info_experiments ORDER BY sap_serp_glean.submission_date)
+    ) AS experiments
+  FROM sap_serp_glean
+  GROUP BY
+    client_id,
+    submission_date
+),
+
+--join sap_serp_glean_aggregates to sap_serp_glean
+sap_serp_glean_aggregates_joined AS (
+  SELECT * FROM sap_serp_glean
+  LEFT JOIN sap_serp_glean_aggregates
+  USING(client_id, submission_date)
+),
+
+-- aggregate by client_id, submission_date, engine, source
+clients_engine_sources_daily AS (
+  SELECT
+
+  --sap stuff
+  client_id,
+  submission_date,
+  channel,
+  app_version,
+  locale,
+  os,
+  os_version,
+  profile_creation_date,
+  session_count,
+  windows_build_number,
+  distribution_id,
+  normalized_channel,
+  country,
+  normalized_os,
+  normalized_os_version,
+  sample_id,
+  profile_group_id,
+  legacy_telemetry_client_id, -- adding this for now so people can join to it if needed
+  experiments,
+  normalized_engine,
+  provider_name,
+  provider_id,
+  partner_code,
+  overridden_by_third_party,
+  search_access_point,
```

Comment:
As noted above, `provider_id`, `partner_code`, `search_access_point` should be from both sap & serp.

Reply:
I believe `provider_id` (aka `provider`) is in the `serp_events` table as [`search_engine`](https://github.com/mozilla/bigquery-etl/blob/d8d41d8a7e34d12fc375d0d262ead4e08e6336ee/sql_generators/serp_events_v2/templates/desktop_query.sql#L72).

I think the `partner_code` should also be added to `serp_events` table if possible - That's an important aspect that we're currently missing.